### PR TITLE
Refactor selective extractor to rule based

### PR DESCRIPTION
**Pull request by ekneg54**
Make selective_extractor rule based.

### CODE REVIEW

**Comment by ekneg54** [comment]
reopen to trigger pipeline

**Review by dtrai2 — CHANGES_REQUESTED** [review]

**Review by dtrai2 — APPROVED** [review]

File: doc/source/user_manual/rule_language.rst [hunk 1]
```diff
@@ -961,6 +961,95 @@ In the following example two files are being used, but only the first existing f
         - PATH_TO_FILE_WITH_LIST
     description: '...'
 
+Selective Extractor
+===================
+
+The selective extractor requires the additional field :code:`selective_extractor`.
+The field :code:`selective_extractor.extract` has to be defined.
+It contains a dictionary of field names and values that should be extracted.
```
Comment (dtrai2) [inline doc/source/user_manual/rule_language.rst:969]:
```suggestion
It contains a dictionary of field names that should be extracted and a target topic to which they should be send to.
```

File: doc/source/user_manual/rule_language.rst [hunk 2]
```diff
@@ -961,6 +961,95 @@ In the following example two files are being used, but only the first existing f
...
+If dot notation is being used, then all fields on the path are being automatically created.
+
+In the following example, the field :code:`extract_test.field.extract` with the value :code:`extracted value` is being extracted
```
Comment (dtrai2) [inline doc/source/user_manual/rule_language.rst:972]: Is the field in the filter automatically the root of the extraction? If not I think it should be like this:

```suggestion
In the following example, the field :code:`field.extract` with the value :code:`extracted value` is being extracted
```

File: doc/source/user_manual/rule_language.rst [hunk 3]
```diff
@@ -961,6 +961,95 @@ In the following example two files are being used, but only the first existing f
...
+and send to the topic :code:`topcic_to_send_to`
+
+..  code-block:: yaml
+    :linenos:
+    :caption: Example with extract from field list
```
Comment (dtrai2) [inline doc/source/user_manual/rule_language.rst:977]:
```suggestion
    :caption: Example rule with extract from field list
```

File: doc/source/user_manual/rule_language.rst [hunk 4]
```diff
@@ -961,6 +961,95 @@ In the following example two files are being used, but only the first existing f
...
+Selective Extractor
+===================
+
+The selective extractor requires the additional field :code:`selective_extractor`.
+The field :code:`selective_extractor.extract` has to be defined.
+It contains a dictionary of field names and values that should be extracted.
+If dot notation is being used, then all fields on the path are being automatically created.
+
+In the following example, the field :code:`extract_test.field.extract` with the value :code:`extracted value` is being extracted
+and send to the topic :code:`topcic_to_send_to`
```
Comment (dtrai2) [inline doc/source/user_manual/rule_language.rst:973]:
```suggestion
and send to the topic :code:`topcic_to_send_to`.
```

File: doc/source/user_manual/rule_language.rst [hunk 5]
```diff
@@ -961,6 +961,95 @@ In the following example two files are being used, but only the first existing f
...
+    }
+
+
+
+Alternatively, the additional field :code:`selective_extractor.extract.extract_from_file` can be added.
+It contains the path to a text file with a list of fields per line to be extracted.
+
+..  code-block:: yaml
+    :linenos:
+    :caption: Example with extract from file
```
Comment (dtrai2) [inline doc/source/user_manual/rule_language.rst:1012]:
```suggestion
    :caption: Example rule with extract from file
```

File: doc/source/user_manual/rule_language.rst [hunk 6]
```diff
@@ -961,6 +961,95 @@ In the following example two files are being used, but only the first existing f
...
+    field2
+    field3
+
+The file has to exist.
+
+It is possible to mix both extraction sources. They will be merged to one list without duplicates.
+
+..  code-block:: yaml
+    :linenos:
+    :caption: Example with extract from file
```
Comment (dtrai2) [inline doc/source/user_manual/rule_language.rst:1035]:
```suggestion
    :caption: Example rule with extract from file
```

**Review by dtrai2 — CHANGES_REQUESTED** [review]

**Review by dtrai2 — APPROVED** [review]

**Pull request merged**